src/test/client/ops.cc

```diff
@@ -0,0 +1,45 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2022 New Dream Network
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#include <iostream>
+#include <errno.h>
+#include "TestClient.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "gtest/gtest-spi.h"
+#include "gmock/gmock-matchers.h"
+#include "gmock/gmock-more-matchers.h"
+
+using ::testing::AnyOf;
+using ::testing::Gt;
+using ::testing::Eq;
+
+TEST_F(TestClient, CheckDummyOP) {
+  int res = client->check_dummy_op(myperm);
+  ASSERT_EQ(res, -EOPNOTSUPP);
+}
+
+TEST_F(TestClient, CheckUnknownSessionOp) {
+  int res = client->check_unknown_session_op();
```

Comment:
I believe you expect the MDS to blocklist the client in this circumstance, no? You should sleep ~2 seconds, then request a new osdmap:

```
objecter->maybe_request_map();
```

(create a helper method in `TestScaffold`) Then add a method to `Client` which queries if the client is blocklisted (get the `blocklisted` variable).

Reply:
By `TestScaffold` you mean `ClientScaffold` class that I created?
Why do we need a new osdmap?

Reply:
The osdmap will contain the blocklist entries.